File: recursive_algorithms/Tower of Hanoi/JavaScript/tower_of_hanoi.js
<script>
// javascript recursive function to
// solve tower of hanoi puzzle
function towerOfHanoi(n, from_rod, to_rod, aux_rod)
{
		if (n == 1)
		{
			document.write("Move disk 1 from rod " + from_rod +
			" to rod " + to_rod+"<br/>");
			return;
		}
		towerOfHanoi(n - 1, from_rod, aux_rod, to_rod);
		document.write("Move disk " + n + " from rod " + from_rod +
		" to rod " + to_rod+"<br/>");
		towerOfHanoi(n - 1, aux_rod, to_rod, from_rod);
	}

	// Driver code
	var n = 4; // Number of disks
	towerOfHanoi(n, 'A', 'C', 'B'); // A, B and C are names of rods

// This code is contributed by gauravrajput1
</script>

File: recursive_algorithms/Tower of Hanoi/C/Tower_Of_Hanoi.c
#include<stdio.h>
void TOH(int n,char x,char y,char z);
void main()
{
    int n;
    printf("Enter the number of disks: ");
    scanf("%d",&n);
    TOH(n,'A','B','C');
}
void TOH(int n,char a,char b,char c)
{
    if(n>0)
    {
        TOH(n-1,a,c,b);
        printf("\n%c -> %c",a,c);
        TOH(n-1,b,a,c);
    }
}